src/LiveComponent/src/Form/Type/MultiStepType.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Silas Joisten <silasjoisten@proton.me>
+ * @author Patrick Reimers <preimers@pm.me>
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+final class MultiStepType extends AbstractType
+{
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setDefault('current_step_name', static function (Options $options): string {
+                return array_key_first($options['steps']);
+            })
+            ->setRequired('steps');
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $options['steps'][$options['current_step_name']]($builder);
```

Comment:
Allowing to use `FormType` as step can be good too. What do you think?

Maybe with something like that (haven't tested it):
```suggestion
        $step = $options['steps'][$options['current_step_name']];
        if (is_callable()) {
            $step($builder);
        } elseif (is_string($step) && is_a($step, FormTypeInterface::class)) {
            $builder->add('step', $step, [
                'inherit_data' => true,
            ]);
        }
```

Reply:
yes i was thinking about the same. so callback or class string of FormTypeInterface.